src/sinks/nats.rs

```diff
@@ -248,22 +259,51 @@ mod integration_tests {
 
     #[tokio::test]
     async fn nats_happy() {
+        // Runs the standard test_publish test below, against a server with no authentication.
+        let subject = format!("test-{}", random_string(10));
+
+        test_publish(
+            subject.clone(),
+            NatsSinkConfig {
+                encoding: EncodingConfig::from(Encoding::Text),
+                connection_name: "".to_owned(),
+                subject: subject.clone(),
+                url: "nats://127.0.0.1:4222".to_owned(),
+                auth: None,
+            },
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn nats_user_password() {
```

Comment:
Would it be worth adding tests for the opposite configurations? That is, authentication to the server that doesn't require it and vice versa? The latter at least would provoke an error.

Reply:
If you think it's worth it I do. Before my changes there was only 1 integration test - I was hesitant to trigger a Cambrian explosion here ;)

Reply:
I think at least one that provokes a failure would be useful, just to double check not passing authentication does in fact fail and so prove our authentication works.